### PR TITLE
Title: Resolved issue with Beacon leader being null

### PR DESCRIPTION
**Pull request by RunTerror**
`Changes Made:`

1. Modified the resolver for the group query.
2. Also, populated the leader field of beacons inside the group.

Screenshot 2024-02-08 at 12 49 27

#374

### CODE REVIEW

**Comment by RunTerror** [comment]
@aadibajpai  Please review my pr

File: graphql/resolvers.js [hunk 1]
```diff
@@ -36,12 +36,20 @@ const resolvers = {
             return beacon;
         },
         group: async (_parent, { id }, { user }) => {
-            const group = await Group.findById(id).populate("leader members beacons");
-            if (!group) return new UserInputError("No group exists with that id.");
-            // return error iff user not in group
-            if (group.leader.id !== user.id && !group.members.includes(user))
-                return new Error("User should be a part of the group");
-            return group;
+            const group = await Group.findById(id).populate('leader members').populate({
```
Comment (aadibajpai) [inline graphql/resolvers.js:39]: looks good, need to review the rest

File: graphql/resolvers.js [hunk 2]
```diff
@@ -73,10 +81,39 @@ const resolvers = {
                     password: await bcrypt.hash(credentials.password, 10),
                 }),
             });
+            console.log(newUser);
             const userObj = await newUser.save();
             return userObj;
         },
 
+        oAuth: async (_parent, { userInput }) => {
```
Comment (aadibajpai) [inline graphql/resolvers.js:89]: what is this for

File: graphql/resolvers.js (hunk 1 above)
Comment (RunTerror) [reply graphql/resolvers.js:39]: Hi @aadibajpai, I'm actively addressing the issue. Additionally, I'm proposing a new feature that allows leaders to toggle between displaying their name or remaining anonymous on the beacon. So i want to close this pr for now soon i'll complete proposed feature and create a new pr.

**Pull request closed without merging**